**Az első 5 feladat ismétlés (többtáblás lekérdezések) a tanulmányi adatbázisból**

1. Készítsünk listát arról, hogy melyik óra melyik teremben van!
    
    1. Jelenítsük meg az óra azonosítóját és a terem nevét!
    2. A listát szűrjük a 3 karakter hosszú, \[100;199\] zárt intervallumba eső termekre! Feltesszük, hogy a 3 karakter hosszú terem nevek csak számjegyeket tartalmaznak.

In [3]:
SELECT o.ora_id,
        t.terem
FROM Orak o JOIN Termek t ON o.terem = t.tkód
WHERE LEN(t.terem) = 3 AND t.terem BETWEEN 100 AND 199

(30 rows affected)

Total execution time: 00:00:00.007

ora_id terem 159 117 160 118 161 118 162 119 163 120 164 119 165 119 166 119 167 120 170 120 172 119 173 119 175 120 176 118 177 119 179 120 180 119 181 118 185 120 192 119 195 116 197 116 198 119 204 120 212 120 218 118 220 119 222 116 226 117 227 116

2. <span style="font-weight: 600; color: rgb(36, 41, 47); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px;">Hány óra van összesen az egyes napokon?</span>

1. Jelenítsük meg a napok nevét és az órák számát az adott napon!
2. Az oszlopokat nevezzük el értelemszerűen!
3. A listát rendezzük óraszám szerint növekvő sorrendbe!

In [ ]:
SELECT
    COUNT(*) AS "Órák száma" /*Count(o.nap is lehetne)*/
FROM Orak o JOIN Napok n ON o.nap = n.nap_id 
GROUP BY n.nap_id 
ORDER by COUNT(*)



3. <span style="font-weight: 600; color: rgb(36, 41, 47); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px;">Készítsünk listát arról, hogy melyik oktatónak hány órája van az Órák táblában!</span>

1. A lista jelenítse meg az oktatók nevét, és az órák számát!
2. Az oszlopokat nevezzük el értelemszerűen!
3. A listából hagyjuk ki az óraadókat!

In [ ]:
--Orak, oktatok és statusok/beosztasok tablak
SELECT okt.nev AS "Név",
        COUNT(o.ora_id) --count(*) is jo, de selectbe ugyanaz legyen mint group by -ban
FROM Orak o JOIN Oktatok okt ON o.tanar = okt.oktato_id
            JOIN statusok s ON okt.[status] = s.Skod
WHERE s.[Status] <> "Óraadó"
GROUP BY okt.nev

4. <span style="color: rgb(36, 41, 47); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px;">Van-e olyan tantárgy, amelyből nincs óra?</span>

1. Jelentsük meg a tantárgy nevét és a tantárgyfelelős nevét!

In [26]:
SELECT t.Tantárgynév,
        okt.nev
FROM Tantárgyak t LEFT JOIN Orak o ON t.Tkód = o.targy  --azt is megmutatja, ahol a bal oldali dolognak nincs párja a jobboldalon
--itt: azt a tárgyat is megjelníti, amihez nincs óra kapcsolva --> null érték

                        JOIN Oktatok okt ON t.Tantárgyfelelős = okt.oktato_id
Where o.targy IS NULL


(1 row affected)

Total execution time: 00:00:00.522

Tantárgynév nev Folyami gáttan Varga Kata

5. <span style="background-color: rgb(255, 255, 255); color: rgb(36, 41, 47); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px;">Hány óra van összesen a 120-as teremben?</span>

**A 6-10 feladatokat halmazműveletek segítségével oldjuk meg, szintén a tanulmányi adatbázison**

6. <span style="background-color:rgb(255, 255, 255);color:rgb(36, 41, 47);font-family:-apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;;font-size:16px;font-weight:600;">Jelenítsük meg azon tanárok azonosítóját és nevét, akik hétfői vagy keddi napokon tanítanak!</span>

7. <span style="background-color: rgb(255, 255, 255); color: rgb(36, 41, 47); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; font-weight: 600;">Melyek azok a tantárgyak, amelyek esetén van óra a 8.00-kor kezdődő sávban, de nincs a 12.30-kor kezdődő sávban?</span>

8. <span style="background-color: rgb(255, 255, 255); color: rgb(36, 41, 47); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; font-weight: 600;">Listázzuk azokat a termeket, amelyekben oktat Kovács László és van olyan tanóra a teremben, ahol a tantárgy nevében nem szerepel az 'ürge' szó!</span>

9. <span style="font-weight: 600; color: rgb(36, 41, 47); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px;">Készítsünk listát arról, hogy melyik tanárnak hány órája van a (munka) hét elején (hétfőn), hét közben (kedd, szerda és csütörtök), illetve a hét végén (péntek)!</span>

1. Jelenítsük meg a tanár azonosítóját, nevét, az időszakot (hét eleje - hét közben - hét vége), valamint az óraszámot!
2. A három megjelölt időszakot külön kérdezzük le, majd fűzzük össze a lekérdezések eredményeit!
3. Rendezzük a listát az oktató neve, azon belül időszak szerint!

10. <span style="background-color: rgb(255, 255, 255); color: rgb(36, 41, 47); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; font-weight: 600;">Melyik napokon nincs órája Pelikán Józsefnek?</span>

In [23]:
-- összes nap

SELECT nap
FROM napok

-- mikor van órája

EXCEPT

SELECT n.nap
FROM Orak o JOIN OKtatok okt ON o.tanar = okt.oktato_id 
            JOIN Napok n ON o.nap = n.nap_id
WHERE okt.nev = 'Pelikán József'

-- megoldás a kettő különbségse



(3 rows affected)

Total execution time: 00:00:00.052

nap Csütörtök Péntek Szerda

**A 11-15 feladatok gyakorlásra szolgálnak a közelgő zh-ra :-)**

11. _Készítsünk listát az oktatók adatairól!_

1. Az oktatók titulusa és neve összefűzve jelenjen meg, közöttük szóközzel!
2. Ha valakinek nincs titulusa, akkor csak a neve jelenjen meg, szóköz nélkül!
3. A listát rendezzük a beosztás kódja, azon belül a státusz kódja szerint növekvő sorrendbe!
4. Az oszlopokat nevezzük el értelemszerűen!

<span style="color: #09885a;">12</span>. Készítsünk listát, amely megmutatja, hogy összesen hány tantárgy van kreditenkénti csoportosításban!

  

A listából hagyjuk ki azokat a tételeket, ahol ez a darabszám <span style="color: #09885a;">1</span>!

A Folyami gáttan tantárgyat ne vegyük figyelembe!

A listát rendezzük darabszám szerint csökkenő sorrendbe!

Az oszlopokat nevezzük el értelemszerűen!

13. _Készítsünk listát az oktatók nevéről, a beosztásuk alapján elvárt és a tényleges óraszámukról!_

1. Az oszlopok neve legyen 'Oktató neve', 'Elvárt óraszám' és 'Tényleges óraszám'
2. Szűrjük a listát azokra az oktatókra, ahol az elvárt és a tényleges óraszám azonos

14. _Készítsünk listát, amely megmutatja, hogy mely tantárgyból nincs óra egyáltalán!_

1. Csak a tantárgy neve jelenjen meg!

15. _Hány órájuk van az óraadóknak napi bontásban?_

1. Csak a napok nevét és az óraszámot jelenítsük meg!
2. Rendezzük a listát óraszám szerint növekvő sorrendbe!